api/apps/api/src/modules/scenarios/planning-units/scenario-planning-units-linker-service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { DbConnections } from '@marxan-api/ormconfig.connections';
+import { ScenariosPlanningUnitGeoEntity } from '@marxan/scenarios-planning-unit';
+import {
+  PlanningUnitGridShape,
+  Project,
+} from '@marxan-api/modules/projects/project.api.entity';
+import { Scenario } from '../scenario.api.entity';
+import { isNil } from 'lodash';
+
+@Injectable()
+export class ScenarioPlanningUnitsLinkerService {
+  constructor(
+    @InjectRepository(
+      ScenariosPlanningUnitGeoEntity,
+      DbConnections.geoprocessingDB,
+    )
+    private readonly puRepo: Repository<ScenariosPlanningUnitGeoEntity>,
+    @InjectRepository(Project)
+    private readonly projectsRepo: Repository<Project>,
+  ) {}
+
+  /** Currently we only support linking regular planning unit geometries
+   * computed via ST_HexagonGrid or ST_SquareGrid */
+  private isPlanningUnitGridShapeSupported(
+    shape: PlanningUnitGridShape,
+  ): boolean {
+    return [
+      PlanningUnitGridShape.hexagon,
+      PlanningUnitGridShape.square,
+    ].includes(shape);
+  }
+
+  /**
+   * Link (existing) planning unit geometries matching project settings and
+   * falling within the project bbox to the scenario being created.
+   */
+  async link(scenario: Scenario): Promise<void> {
+    const project = await this.projectsRepo.findOneOrFail(scenario.projectId);
+    if (
+      !this.isPlanningUnitGridShapeSupported 
```

Comment:
it's always false, isn't it? (`callme()`)

Reply:
well spotted, that was completely unfinished, just happened to please tsc in its form 😬

Reply:
fixed (actually implemented) in d168393b1bb033fbe88a18dfa0140c65726f14f6